crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs

```diff
@@ -200,6 +222,40 @@ impl Drop for Context {
     }
 }
 
+async fn create_new_erc20msg_client(
+    deploy_key: &str,
+    endpoint: &str,
+    erc20_messaging_contract_address: Address,
+) -> Result<IERC20MessagingClient, Box<dyn std::error::Error>> {
+    let wallet: LocalWallet = deploy_key.parse()?;
+    let http_provider =
+        Provider::<Http>::try_from(endpoint)?.interval(std::time::Duration::from_millis(20u64));
+    let chain_id = http_provider.get_chainid().await?;
+    let client = Arc::new(SignerMiddleware::new(
+        http_provider,
+        wallet.clone().with_chain_id(chain_id.as_u64()),
+    ));
+    let i_erc20_messaging = IERC20Messaging::new(erc20_messaging_contract_address, client);
+    Ok(i_erc20_messaging)
+}
+
+async fn create_new_erc20_client(
```

Comment:
This method is really similar to the one above, maybe we can merge or share some part of it

Reply:
Ok, I'll work a lot more on the sequencer tests in the foreseeable future so I'll optimize it